File: TopCoder/src/main/java/com/leetcode/integer/PalindromNumber.java
package com.leetcode.integer;

public class PalindromNumber {

    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int originalNumber = x;
        int reversedNumber = 0;

        while ( x > 0) {
            reversedNumber = reversedNumber*10 + x%10;
            x = x/10;
        }
        return reversedNumber == originalNumber;
    }

    public static void main(String[] args) {
        int number = 121;
        boolean res = new PalindromNumber().isPalindrome(number);
        System.out.println(res);
    }
}

File: TopCoder/python/arr/PalindromNumber.py
class Solution:
    def isPalindrome(self, x: 'int') -> 'bool':
        original = x
        reversedNumber = 0
        if x < 0 :
            return False

        while x > 0:
            reversedNumber = (reversedNumber * 10) + int((x % 10))
            x = int(x / 10)

        return reversedNumber == original

sol = Solution()
res = sol.isPalindrome(121)
print(res)